pandas/tests/groupby/test_bin_groupby.py

```diff
@@ -27,6 +27,23 @@ def test_series_grouper():
     tm.assert_almost_equal(counts, exp_counts)
 
 
+def test_series_grouper_timestamp():
+    # GH 42390
+    obj = Series([1], index=[pd.Timestamp("2018-01-16 00:00:00+00:00")], dtype=np.intp)
+    labels = np.array([0])
+
+    def agg(series):
+        # this should not raise
+        if series.isna().values.all():
+            return None
+        return np.sum(series)
+
+    grouper = libreduction.SeriesGrouper(obj, agg, labels, 1)
+    result, counts = grouper.get_result()
```

Comment:
pls use assert_series_equal

Reply:
Hi, you mean `assert_equal` ? Because `result` and `counts` here are not `Series` but `np.array`. I modified to use `assert_equal` instead.

Reply:
assert_numpy_array_equal

pls let the question-asker hit the "resolve conversation" button

Reply:
Ok. I now use assert_numpy_array_equal.

Reply:
more to the point also do a full on test of the original example (the `.agg`)

Reply:
> more to the point also do a full on test of the original example (the .agg)

@philpep can you test the user-facing function here